pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -112,14 +107,28 @@ def run(self, batch: Any, batch_idx: int, dataloader_idx: int) -> AttributeDict:
         super().run(batch, batch_idx, dataloader_idx)
         output = AttributeDict(signal=0, training_step_output=self.batch_outputs)
         self.batch_outputs = None  # free memory
+
         return output
 
-    def reset(self) -> None:
+    def _initialize(self):
         """Resets the loop state"""
         self._hiddens = None
         self.batch_idx = 0
         self.batch_outputs = [[] for _ in range(len(self.trainer.optimizers))]
 
+    def restore(self) -> bool:
+        """Restore the loop state"""
+        self._initialize()
+
+        # restarting isn't finished yet.
+        return False
+
+    def reset(self) -> None:
+        """Resets the loop state"""
+        self._initialize()
+
+        self.optim_progress.optimizer.reset_on_epoch()
```

Comment:
these _initialize() calls could be simplified as proposed in the earlier comment